### ISSUE TITLE
lack of module inter_attn()

### ISSUE THREAD

**Issue by zkjisj** [CVer-Yang/CCNet#2]
Thank for your contribution of the opensource, but I meet the question of lack the module inter_attn() when I want to use your model. I encounter this problem in this following line: 
https://github.com/CVer-Yang/CCNet/blob/ab030e0ec2557402f79fcd73f2a9fa6868e3c718/CCNet.py#L282
Looking for your help!

**Comment by CVer-Yang** [comment]
您好，感谢关注。 
class inter_attn(nn.Module):
    def __init__(self, f_dim, n_heads=8, d_q=None, d_v=None, dropout=0.3):
        super().__init__()

        self.build_inter_attn(f_dim, n_heads, d_q, d_v, dropout)

        for m in self.modules():
            weights_init(m)

    def build_inter_attn(self, f_dim, n_heads=4, d_q=None, d_v=None, dropout=0.1):
        if d_q is None:
            d_q = f_dim // n_heads
        if d_v is None:
            d_v = f_dim // n_heads

        self.n_heads = n_heads
        self.d_q = d_q
        self.d_v = d_v
        self.norm = d_q ** 0.5
        self.f_dim = f_dim

        self.dropout1 = nn.Dropout(dropout)
        self.dropout2 = nn.Dropout(dropout)
        self.w_qs = nn.Linear(f_dim, n_heads * d_q)
        self.w_ks = nn.Linear(f_dim, n_heads * d_q)
        self.w_vs = nn.Linear(f_dim, n_heads * d_v)
        self.fc = nn.Linear(n_heads * d_v, f_dim)

        self.layer_norm1 = nn.LayerNorm(f_dim, eps=1e-6)
        self.layer_norm2 = nn.LayerNorm(f_dim, eps=1e-6)

    def inter_attn(self, Local, Fuse, mask_L2R=None, mask_R2L=None):
        b, c, h, w = Local.shape
        Lf = Local.view(b, c, -1).permute(0, 2, 1)
        Rf = Fuse.view(b, c, -1).permute(0, 2, 1)
        BS, V, fdim = Lf.shape
        assert fdim == self.f_dim
        BS, V, fdim = Rf.shape
        assert fdim == self.f_dim

        Lf2 = self.layer_norm1(Lf)
        Rf2 = self.layer_norm2(Rf)

        Lq = self.w_qs(Lf2).view(BS, V, self.n_heads, self.d_q).transpose(1, 2)  # BS x h x V x q
        Lk = self.w_ks(Lf2).view(BS, V, self.n_heads, self.d_q).transpose(1, 2)  # BS x h x V x q
        Lv = self.w_vs(Lf2).view(BS, V, self.n_heads, self.d_v).transpose(1, 2)  # BS x h x V x v

        Rq = self.w_qs(Rf2).view(BS, V, self.n_heads, self.d_q).transpose(1, 2)  # BS x h x V x q
        Rk = self.w_ks(Rf2).view(BS, V, self.n_heads, self.d_q).transpose(1, 2)  # BS x h x V x q
        Rv = self.w_vs(Rf2).view(BS, V, self.n_heads, self.d_v).transpose(1, 2)  # BS x h x V x v

        attn_R2L = torch.matmul(Lq, Rk.transpose(-1, -2)) / self.norm  # bs, h, V, V
  

        
        if mask_R2L is not None:
            attn_R2L = attn_R2L.masked_fill(mask_R2L == 0, -1e9)

        attn_R2L = F.softmax(attn_R2L, dim=-1)  # bs, h, V, V
      

        attn_R2L = self.dropout1(attn_R2L)


        feat_R2L = torch.matmul(attn_R2L, Rv).transpose(1, 2).contiguous().view(BS, V, -1)

        feat_R2L = self.dropout2(self.fc(feat_R2L))

        Lf = Lf + feat_R2L


        Local = Lf.view(Local.size(0), Local.size(2), Local.size(3), Local.size(1)).permute(0, 3, 1, 2)

        return Local

    def forward(self, Lf, Rf, mask_L2R=None, mask_R2L=None):

        Lf = self.inter_attn(Lf, Rf, mask_L2R, mask_R2L)

        return Lf

**Comment by zkjisj** [comment]
Thanks for your kindly reply, but I can't find the use of this module in the this CCNet.py file. Is this module necessary for the model?
